### PR TITLE
[PF-220] Add metrics to record handout request by pool id

### PR DESCRIPTION
**Pull request by yonghaoy**

### CODE REVIEW

File: src/main/java/bio/terra/buffer/common/MetricsHelper.java [hunk 1]
```diff
@@ -36,6 +36,10 @@ private MetricsHelper() {}
           "ready resource count to pool size ratio",
           RESOURCE_TO_POOL_SIZE_RATIO);
 
+  private static final Measure.MeasureLong HANDOUT_RESOURCE_COUNT =
```
Comment (wchamber) [inline src/main/java/bio/terra/buffer/common/MetricsHelper.java:39]: Why do we need this? Don't we already have this information from the RESOURCE_STATE_COUNT?
As implementdd, this actually counts the number of handout *requests*; if I use the same handout id in two requests, they will both increment this count.

**Review by wchamber — CHANGES_REQUESTED** [review]

File: src/main/java/bio/terra/buffer/common/MetricsHelper.java (hunk 1 above)
Comment (yonghaoy) [reply src/main/java/bio/terra/buffer/common/MetricsHelper.java:39]: This is what Greg suggested from a 1 on 1: to have a metrics to track the handout request event, that would be a good metrics to measure pool qps stuff..

Comment (gjuggler) [reply src/main/java/bio/terra/buffer/common/MetricsHelper.java:39]: My thought was that having an event-style metric might be a useful complement to the existing gauge-style metrics. This PR seems to follow the same pattern as the existing counts. Is there no underlying difference in how OpenCensus tracks gauges vs. events?

Comment (wchamber) [reply src/main/java/bio/terra/buffer/common/MetricsHelper.java:39]: I believe the difference between gauges and events is in the View, `Aggregation.LastValue` vs `Aggregation.Count`. https://opencensus.io/stats/view/#aggregations

Greg, to be clear, do you want an event of the handout requests, handout events, or something else? Are we explicitly duplicating the information in another form for convenience, or measuring something slightly different?

**Comment by yonghaoy** [comment]
@wchamber @gjuggler Moved metrics to controller. PTAL. Thanks

File: src/main/java/bio/terra/buffer/common/MetricsHelper.java (hunk 1 above)
Comment (wchamber) [inline src/main/java/bio/terra/buffer/common/MetricsHelper.java:39]: I think this, and it's metric description, and the method name, comments, etc. should be updated to make it clear this is counting HANDOUT_RESOURCE_REQUEST_COUNT, - requests & not handouts. worth a comment on the distinction.

**Review by wchamber — CHANGES_REQUESTED** [review]

File: src/main/java/bio/terra/buffer/common/MetricsHelper.java (hunk 1 above)
Comment (yonghaoy) [reply src/main/java/bio/terra/buffer/common/MetricsHelper.java:39]: Changed view name ,counter name and method name.

**Review by wchamber — APPROVED** [review]

**Pull request merged**